src/Compilers/CSharp/Test/Emit2/FlowAnalysis/FlowDiagnosticTests.cs

```diff
@@ -2759,5 +2759,27 @@ struct Empty
                 Diagnostic(ErrorCode.ERR_ImplicitlyTypedOutVariableUsedInTheSameArgumentList, "x").WithArguments("x").WithLocation(6, 25)
                 );
         }
+
+        [Theory]
+        [InlineData("object", "new {}")]
+        [InlineData("dynamic", "new {}")]
+        [InlineData("System.ValueType", "((int)x0)/2")]
+        public void ConstantsExpectedInPatternExpression(string type, string expression)
```

Comment:
>ConstantsExpectedInPatternExpression [](http://example.com/codeflow?start=20&length=36)

The placement of this test is somewhat unexpected. Consider moving it to a file/type dedicated to patterns.